### PR TITLE
Updated Tag World filtering.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by Wild1145** [comment]
@marcocorriero What issue does this resolve?

**Comment by marcocorriero** [comment]
@Wild1145 i added new default admins tags in the tag filtering

**Comment by JeromSar** [comment]
@marcocorriero Issue number _(i.e. #xxx)_ please.

**Comment by OxLemonxO** [comment]
This needs formatting.

**Comment by marcocorriero** [comment]
#1588

**Comment by notanoperator** [comment]
possibly also add "mod" for a shortened version of "moderator"

**Comment by Wild1145** [comment]
You need to correct the code formatting.

**Comment by Wild1145** [comment]
Thank you for contributing to the project, could you please amend your opening message to include the issue that this PR resolves. Thank You.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 1]
```diff
@@ -15,145 +15,96 @@
...
-
-                for (final Player player : server.getOnlinePlayers())
-                {
-                    final FPlayer playerdata = plugin.pl.getPlayer(player);
-                    if (playerdata.getTag() != null)
-                    {
-                        msg(player.getName() + ": " + playerdata.getTag());
+    public static final List<String> FORBIDDEN_WORDS;
+    
+    public boolean run(final CommandSender sender, final Player playerSender, final Command cmd, final String commandLabel, final String[] args, final boolean senderIsConsole) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:20]: Formatting needs changing here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 2]
```diff
@@ -15,145 +15,96 @@
...
+        if (args.length == 1) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:21]: Formatting is also incorrect here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 3]
```diff
@@ -15,145 +15,96 @@
...
+            if ("list".equalsIgnoreCase(args[0])) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:22]: Formatting is also incorrect here...

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 4]
```diff
@@ -15,145 +15,96 @@
...
+                this.msg("Tags for all online players:");
+                for (final Player player : this.server.getOnlinePlayers()) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:24]: You need to correct the formatting here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 5]
```diff
@@ -15,145 +15,96 @@
...
+                    final FPlayer playerdata = ((TotalFreedomMod)this.plugin).pl.getPlayer(player);
+                    if (playerdata.getTag() != null) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:26]: Also formatting here needs changing.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 6]
```diff
@@ -15,145 +15,96 @@
...
                 }
-
                 return true;
             }
-            else if ("clearall".equalsIgnoreCase(args[0]))
-            {
-                if (!plugin.al.isAdmin(sender))
-                {
-                    noPerms();
+            if ("clearall".equalsIgnoreCase(args[0])) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:32]: Please correct the formatting here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 7]
```diff
@@ -15,145 +15,96 @@
...
+                if (!((TotalFreedomMod)this.plugin).al.isAdmin(sender)) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:33]: Also please correct the formatting here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 8]
```diff
@@ -15,145 +15,96 @@
...
-
                 int count = 0;
-                for (final Player player : server.getOnlinePlayers())
-                {
-                    final FPlayer playerdata = plugin.pl.getPlayer(player);
-                    if (playerdata.getTag() != null)
-                    {
-                        count++;
-                        playerdata.setTag(null);
+                for (final Player player2 : this.server.getOnlinePlayers()) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:39]: The formatting here also needs changing.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 9]
```diff
@@ -15,145 +15,96 @@
...
+                    final FPlayer playerdata2 = ((TotalFreedomMod)this.plugin).pl.getPlayer(player2);
+                    if (playerdata2.getTag() != null) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:41]: Formatting is also out here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 10]
```diff
@@ -15,145 +15,96 @@
...
-            {
-                if (senderIsConsole)
-                {
-                    msg("\"/tag off\" can't be used from the console. Use \"/tag clear <player>\" or \"/tag clearall\" instead.");
-                }
-                else
-                {
-                    plugin.pl.getPlayer(playerSender).setTag(null);
-                    msg("Your tag has been removed.");
+            else {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:49]: Formatting is off here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 11]
```diff
@@ -15,145 +15,96 @@
...
+                if ("off".equalsIgnoreCase(args[0])) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:50]: Formatting needs changing here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 12]
```diff
@@ -15,145 +15,96 @@
...
+                    if (senderIsConsole) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:51]: Formatting and additional whitespace needs looking at here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 13]
```diff
@@ -15,145 +15,96 @@
...
+                        this.msg("\"/tag off\" can't be used from the console. Use \"/tag clear <player>\" or \"/tag clearall\" instead.");
+                    }
+                    else {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:54]: Whitespace and formatting needs checking here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 14]
```diff
@@ -15,145 +15,96 @@
...
             }
         }
-        else if (args.length >= 2)
-        {
-            if ("clear".equalsIgnoreCase(args[0]))
-            {
-                if (!plugin.al.isAdmin(sender))
-                {
-                    noPerms();
+        else {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:63]: Formatting needs changing here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 15]
```diff
@@ -15,145 +15,96 @@
...
+            if (args.length < 2) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:64]: Formatting & White space check here please.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 16]
```diff
@@ -15,145 +15,96 @@
...
+                return false;
+            }
+            if ("clear".equalsIgnoreCase(args[0])) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:67]: Formatting check here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 17]
```diff
@@ -15,145 +15,96 @@
...
-                        msg("That tag is too long (Max is 20 characters).");
-                        return true;
-                    }
-
-                    for (String word : FORBIDDEN_WORDS)
-                    {
-                        if (rawTag.contains(word))
-                        {
-                            msg("That tag contains a forbidden word.");
+            else {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:81]: Formatting is out here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 18]
```diff
@@ -15,145 +15,96 @@
...
+                if ("set".equalsIgnoreCase(args[0])) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:82]: Please correct the formatting here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 19]
```diff
@@ -15,145 +15,96 @@
...
+                    final String inputTag = StringUtils.join((Object[])args, " ", 1, args.length);
+                    final String outputTag = FUtil.colorize(StringUtils.replaceEachRepeatedly(StringUtils.strip(inputTag), new String[] { "ยง", "&k" }, new String[] { "", "" })) + ChatColor.RESET;
+                    if (!((TotalFreedomMod)this.plugin).al.isAdmin(sender)) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:85]: Please also correct the formatting here

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 20]
```diff
@@ -15,145 +15,96 @@
...
+                        final String rawTag = ChatColor.stripColor(outputTag).toLowerCase();
+                        if (rawTag.length() > 20) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:87]: Please correct the formatting here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 21]
```diff
@@ -15,145 +15,96 @@
...
+                            this.msg("That tag is too long (Max is 20 characters).");
                             return true;
                         }
+                        for (final String word : Command_tag.FORBIDDEN_WORDS) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:91]: Formatting is off here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 22]
```diff
@@ -15,145 +15,96 @@
...
+                            if (rawTag.contains(word)) {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:92]: The formatting is also off here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 23]
```diff
@@ -15,145 +15,96 @@
...
                 return false;
             }
         }
-        else
-        {
-            return false;
-        }
+    }
+    
+    static {
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:107]: Please correct this formatting.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java [hunk 24]
```diff
@@ -15,145 +15,96 @@
...
+        FORBIDDEN_WORDS = Arrays.asList("admin", "owner", "moderator", "developer", "console","MOD", "CONSOLE", "SA", "STA", "SrA");
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_tag.java:108]: This would be better off being set in the config and not hard coded.

**Review by Wild1145 — CHANGES_REQUESTED** [review]
Please review and correct the changes requested.

**Pull request closed without merging**